File: 2-Patron/python/FabricaDeMascotas.py
import FabricaDeCriaturas
import MascotaGrande
import MascotaPequeña


class FabricaDeMascotas(FabricaDeCriaturas):
    def crearGrande(self, nombre):
        return MascotaGrande(nombre, 100, 1000, 1)

    def crearPequeña(self, nombre):
        return MascotaPequeña(nombre, 10, 100, 1)

File: 2-Patron/Java/FabricadeMascotas.java
package abstractmasterwhole;
public class FabricadeMascotas implements FabricadeCriaturas {

    @Override
    public Criatura crearGrande(String nombre) {
        return new MascotaGrande(nombre, 100, 1000, 1);
    }

    @Override
    public Criatura crearPequeña(String nombre){ 
            return new MascotaPequeña(nombre, 10, 100, 1);
    }

    
    
}
